src/UserSpaceInstrumentation/ExecuteMachineCode.cpp

```diff
@@ -49,18 +49,25 @@ ErrorMessageOr<uint64_t> ExecuteMachineCode(MemoryInTracee& code_memory, const M
 
   RegisterState registers_for_execution = original_registers;
   registers_for_execution.GetGeneralPurposeRegisters()->x86_64.rip = code_memory.GetAddress();
-  // The calling convention for x64 assumes the 128 bytes below rsp to be usable as a scratch pad
-  // for the current function. This area is called the 'red zone'. The function we interrupted might
-  // have stored temporary data in the red zone and the function we are about to execute might do
-  // the same. To keep them separated we decrement rsp by 128 bytes.
-  // The calling convention also asks for rsp being to be aligned to 16 bytes so we additionally
-  // round down to the previous multiple of 16.
   const uint64_t old_rsp = original_registers.GetGeneralPurposeRegisters()->x86_64.rsp;
-  constexpr int kSizeRedZone = 128;
+  // The System V calling convention (Linux x64) assumes the 128 bytes below rsp to be usable as a
```

Comment:
I expected to find a different implementation of `ExecuteMachineCode` for the Windows calling convention. But I agree that this can be done in the same method and this saves us some duplication. The only concern left is performance, but I checked this is not in any sensitive execution path. So I really think this is fine.

Reply:
Why do you think the changes here could affect performance? The only difference is that we decrement RSP by 160 instead of 128. Maybe I missed something?

Reply:
Oh, right, for some reason I assumed that the allocation of the shadow space involved something else beyond incrementing the stack pointer. That was rubbish. Please disregard.